### PR TITLE
Tests for attach/detach operation with option matrix.

### PR DESCRIPTION
**Pull request by rbian**

### CODE REVIEW

**Comment by waynesun09** [comment]
saw lots of fail with change media matrix, please check the internal CI jobs for this.

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
+import logging
+import time
+from autotest.client.shared import error
+from autotest.client.shared import utils
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py]: new cases have been switch to avocado utils:
from avocado.utils import process

File: libvirt/tests/src/virsh_cmd/domain/virsh_update_device_matrix.py [hunk 2]
```diff
@@ -0,0 +1,432 @@
+import os
+import time
+import shutil
+import logging
+from autotest.client.shared import error
+from autotest.client.shared import utils
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_update_device_matrix.py]: also here

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py (hunk 1 above)
Comment (rbian) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py]: Thanks, I will update it.

**Comment by waynesun09** [comment]
ACK
The matrix is huge and testing result on x86 CI is great, so I'm gonna merge this now, thanks.

**Pull request merged**